File: Leetcode/19-RemoveNthNodeFromEndOfList/19-RemoveNthNodeFromEndOfList.java
// Last updated: 6/7/2026, 11:28:01 PM
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) 
    {
        ListNode temp = head;
        int l = 0;
        while(temp!=null)
        {
            l++;
            temp=temp.next;
        }
        System.out.println("l is "+l);
        temp = head;
        int c=0;
        if(n!=0)
        {
            while(c<l-n-1)
            {
                c++;
                temp=temp.next;
            }
        }
        else
        {
            while(c<l-n-1)
            {
                c++;
                temp=temp.next;
            }
        }
        if(l==n)
        {
            return head.next;
        }
        System.out.println("temp val is "+temp.val);
        temp.next=temp.next.next;
        return head;
    }
}

File: Leetcode/19-RemoveNthNodeFromEndOfList/19-RemoveNthNodeFromEndOfList.cpp
// Last updated: 6/7/2026, 11:27:56 PM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int N) {
        if(head==nullptr)
    {
        return head;
    }
    int n = 0;
    ListNode* cur = head;
    while(cur!=nullptr)
    {
        n++;
        cur = cur->next;
    }
    int k = n-N;
    if(k==0)
    {
        return head->next;
    }
    else
    {
        cout<<" k is "<<k<<"\n";
    cur = head;
    for(int i = 1;i<=k-1;i++)
    {
        cur = cur->next;
    }
    cout<<cur->val<<"\n";
    cur->next=cur->next->next;
    return head;
    }
    }
};